arroyo/processing/strategies/dead_letter_queue/policies/produce.py

```diff
@@ -0,0 +1,85 @@
+import json
+import time
+from collections import deque
+from concurrent.futures import Future
+from datetime import datetime
+from typing import Any, Deque, Mapping, Optional
+
+from arroyo.backends.abstract import Producer
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessage,
+    InvalidMessages,
+    JSONSerializable,
+)
+from arroyo.types import Message, Topic
+from arroyo.utils.codecs import Encoder
+from arroyo.utils.metrics import get_metrics
+
+MAX_QUEUE_SIZE = 5000
+DATE_TIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+class JSONMessageEncoder(Encoder[bytes, Mapping[str, JSONSerializable]]):
```

Comment:
Since the InvalidMessage is already doing part of the encoding (to a dict), I wonder if we should remove this encoder and just move this code into InvalidMessage as well so all encoding is happening in one spot.